scipy/linalg/tests/test_lapack.py

```diff
@@ -1743,26 +1743,22 @@ def test_syequb():
         assert_equal(np.log2(s).astype(int), desired_log2s)
 
 
-@pytest.mark.skip(reason="This fails inexplicably when some ?gejsv tests "
-                  "run. See gh-11902.")
-@pytest.mark.parametrize("dtype", COMPLEX_DTYPES)
-def test_heequb(dtype):
-    desired_log2s = np.array([[-2, -7, -2, -4, -2, -3, -2, -2, -1, -2],
-                              [1, -10, 0, -6, -1, -4, -1, -2, -1, -2]])
-    heequb = get_lapack_funcs('heequb', dtype=dtype)
-
-    d = np.array([dtype(1j) * 2**x for x in range(-5, 5)], dtype=dtype)
-    A = np.diag(d)
-    subdiags = np.array([dtype(1j) * 2**(9-x) for x in range(9)],
-                        dtype=dtype)
-    A[range(1, 10), range(0, 9)] = subdiags
-    s, scond, amax, info = heequb(A, lower=1)
+def test_heequb():
+    # zheequb has a bug for versions =< LAPACK 3.9.0
+    # See Reference-LAPACK gh-61 and gh-408
+    # Hence the zheequb test is customized accordingly to avoid
+    # work scaling.
+    A = np.diag([2]*5 + [1002]*5) + np.diag(np.ones(9), k=1)*1j
+    s, scond, amax, info = lapack.zheequb(A)
+    assert_equal(info, 0)
+    assert_allclose(np.log2(s), [0., -1.]*2 + [0.] + [-4]*5)
 
+    A = np.diag([2**abs(x)+0j for x in range(-5, 6)])
```

Comment:
```suggestion
    A = np.diag([2**abs(x)+0j for x in range(-5, 6)])
    A = np.diag(2**np.abs(np.arange(-5,6)) + 0j)
```
NBD, just showing that I'm taking a look : )